### DISCUSSION TITLE
I'm using an HashTable inside a PodeState. There is a way to remove a key from the PodeState?

### DISCUSSION THREAD

**Discussion by mdaneri** [Badgerati/Pode#1129] [Q&A]
Hi gents,
I have a question regarding PodeState. 
Everything working great but there is a thing that I want to implement, and I have some troubles.
I want to remove a key with the associated value
I create the podState with a code similar to this 
```
  $Scripts = (Get-PodeState -Name 'Data') 
  $Scripts.scripts += @{"$($Script.id)" = $Script } 
  # save the state to file
   Save-PodeState -Path $using:ScripsJsonData 
```

what I want to do is to remove one element from $Scripts.scripts

**Comment by Badgerati** [comment]
Hi @mdaneri,

It appears that `$Scripts.scripts` is a hashtable? If so, then if you know the `$Script.id` the following will remove an item:

```powershell
$Scripts.scripts.Remove($Script.id)
```

**Reply by mdaneri** [reply]
doesn't work 
Category: System.Private.CoreLib
Message: Collection was of a fixed size.
StackTrace:    at System.Array.System.Collections.IList.Remove(Object value)
   at CallSite.Target(Closure, CallSite, Object, Object)

**Comment by mdaneri** [comment]
I solved the mystery. I was initializing the hashtable like an array. 
it had a strange structure.   
` Set-PodeState -Name 'Data' -Value @{'scripts' = @(); 'scriptId' = 0 } | Out-Null`
instead of 
 `Set-PodeState -Name 'Data' -Value @{'scripts' = @{}; 'scriptId' = 0 } | Out-Null`
